### DISCUSSION TITLE
仿照DialogWithAction写了一个登录窗体，但是Dialog再弹窗就无效了，也不报错。

### DISCUSSION THREAD

**Discussion by YSW33** [irihitech/Ursa.Avalonia#171] [Q&A]
![image](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/37716279-2114-48f6-8dcc-921b1260bb35)
![image](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/59c3bd3f-6184-4d77-8535-5f58695ad231)
![image](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/5dc43603-23ea-4977-9a2f-263af5701d29)

**Comment by rabbitism** [comment]
是不是忘记指定host id了？

**Comment by YSW33** [comment]
> 是不是忘记指定host id了？

我看DEMO里面并没有指定host id，直接就是OverlayDialog.ShowCustomModal<DialogWithAction, DialogWithActionViewModel, bool>(new DialogWithActionViewModel());

**Comment by rabbitism** [comment]
> > 是不是忘记指定host id了？
> 
> 我看DEMO里面并没有指定host id，直接就是OverlayDialog.ShowCustomModal<DialogWithAction, DialogWithActionViewModel, bool>(new DialogWithActionViewModel());

没有ID就意味着我在使用放在MainView的那个无ID host。你这个case我不确定是不是放了那个host。

**Comment by YSW33** [comment]
> > > 是不是忘记指定host id了？
> > 
> > 
> > 我看DEMO里面并没有指定host id，直接就是OverlayDialog.ShowCustomModal<DialogWithAction, DialogWithActionViewModel, bool>(new DialogWithActionViewModel());
> 
> 没有ID就意味着我在使用放在MainView的那个无ID host。你这个case我不确定是不是放了那个host。

确实是MainView，我在MainView加上就好了，name是host，这个是默认的ID吗？

**Comment by rabbitism** [comment]
> 确实是MainView，我在MainView加上就好了，name是host，这个是默认的ID吗？

DialogHost有一个属性叫做HostId，这个是在ShowDialog时候需要指定的东西，如果这个没设置，他的Id就是null

**Comment by rabbitism** [comment]
要不你把这个demo上传github我看看是什么情况

**Comment by YSW33** [comment]
> 要不你把这个demo上传github我看看是什么情况

https://github.com/YSW33/Test.git

**Comment by rabbitism** [comment]
![image](https://github.com/irihitech/Ursa.Avalonia/assets/14807942/7b52f6e5-c454-4dd4-9763-5a4f9ee78e4b)
我运行起来没什么问题（不过你的demo里面Plugin项目缺少dependency，我自己补上了）

**Comment by YSW33** [comment]
> ![image](https://private-user-images.githubusercontent.com/14807942/314466533-7b52f6e5-c454-4dd4-9763-5a4f9ee78e4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5MzQ3MDQsIm5iZiI6MTcxMDkzNDQwNCwicGF0aCI6Ii8xNDgwNzk0Mi8zMTQ0NjY1MzMtN2I1MmY2ZTUtYzQ1NC00ZGQ0LTk3NjMtNWE0ZjllZTc4ZTRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDExMzMyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhMzRlM2EwNjIyNTMzNGM3OWVjYjczYzIxMTE3ZjdmZDM1ZTlmNzJjOWU1YjNhNjI5NjgwMmE3MDgzMzliZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wSA3pamA-Ib_Lzq-RMT3CD0U3p4ucuu4dNPwvM127h4) 我运行起来没什么问题（不过你的demo里面Plugin项目缺少dependency，我自己补上了）

InitializeComponent也不提示错误吗？

**Comment by rabbitism** [comment]
> InitializeComponent也不提示错误吗？

有可能就是因为你的plugin没有reference ursa

**Comment by YSW33** [comment]
> > InitializeComponent也不提示错误吗？
> 
> 有可能就是因为你的plugin没有reference ursa

我试着把AVALONIA降到11.0.7  InitializeComponent 就正常了，反之升级到11.0.10就不行。

**Comment by rabbitism** [comment]
所以我可以认为这个问题解决了吗

**Comment by YSW33** [comment]
> 所以我可以认为这个问题解决了吗

也算是解决了吧，估计后面的版本可能是因为哪个引用造成的。
